### ISSUE TITLE
Distribution of scrape targets when number of shards change

### ISSUE THREAD

**Issue by champak** [prometheus-operator/prometheus-operator#6196]
Hi folks,

I have a question regarding expected behavior when the number of Prometheus shards change. E.g. consider 300 pod targets that were being scraped by 3 shards with around 100 targets getting scraped per shard. If the number of shards are changed to say 2, should we expect the number of targets scraped per shard to go to around 150? The documentation at https://github.com/prometheus-operator/prometheus-operator/blob/d52d58d90dcc8df3665147941ecad3e918ca208c/Documentation/user-guides/shards-and-replicas.md?plain=1
 states:
"
Note that scaling down shards will not reshard data onto remaining instances. It must be manually moved. Increasing shards will not reshard data either. 
"

In our testing we seem to see that when the number of shards changes, after a couple of minutes the pod targets are rebalanced across the new number of shards. It seems as though once the Operator has a chance to go through the reconciliation loop, it recomputes the spread of targets across shards. This seems to contradict what the documentation says. Having the scrape targets automatically rebalanced would of course be a great feature to have. Thanks for any pointers!

**Comment by simonpasquier** [comment]
The documentation talks about the data already being stored which will not be automatically moved to the remaining instances. You might want to follow https://github.com/prometheus-operator/prometheus-operator/issues/4967 and https://github.com/prometheus-operator/prometheus-operator/pull/6198.

**Comment by champak** [comment]
@simonpasquier thanks very much for your response! If I followed right, scaling up should not have any data loss impact since new data will get sharded across the new set of instances. As long as querying the data is across data-sets aggregated from all of the shard instances, query results should be consistent. Scaling down could result in data loss unless manual steps to restore data are taken since storage associated with scaled down instances could get reaped. Does that sound right?

 In our case we are using Victoria Metrics for long term storage (Prometheus remote write to VM). Since data from a scaled down shard is already going to be stored in VM, I am expecting historical data will stay around for querying from VM.

**Comment by simonpasquier** [comment]
You're correct. If you're only relying on Prometheus for scraping and exporting metrics (no local storage/query) then you should be fine.

**Comment by champak** [comment]
@simonpasquier thanks for confirming!
